pinot-controller/src/main/java/org/apache/pinot/controller/util/TableTierReader.java

```diff
@@ -86,10 +91,23 @@ public TableTierDetails getTableTierDetails(String tableNameWithType, @Nullable
       List<String> expectedSegmentsOnServer = entry.getValue();
       TableTierInfo tableTierInfo = serverToTableTierInfoMap.get(server);
       for (String expectedSegment : expectedSegmentsOnServer) {
-        tableTierDetails._segmentTiers.computeIfAbsent(expectedSegment, (k) -> new HashMap<>()).put(server,
+        tableTierDetails._segmentCurrentTiers.computeIfAbsent(expectedSegment, (k) -> new HashMap<>()).put(server,
             (tableTierInfo == null) ? ERROR_RESP_NO_RESPONSE : getSegmentTier(expectedSegment, tableTierInfo));
       }
     }
+    if (!includeTargetTier) {
```

Comment:
any downside of removing the flag and just making this the default behavior to include target tier info?